spot-contracts/contracts/BondIssuer.sol

```diff
@@ -140,17 +180,30 @@ contract BondIssuer is IBondIssuer, OwnableUpgradeable {
     /// @dev Lazily issues a new bond when the time is right.
     function getLatestBond() external override returns (IBondController) {
         issue();
-        // NOTE: The latest bond will be at the end of the list.
-        return IBondController(_issuedBonds.at(_issuedBonds.length() - 1));
+        return _lastBond;
     }
 
     /// @inheritdoc IBondIssuer
     function issuedCount() external view override returns (uint256) {
-        return _issuedBonds.length();
+        return _activeBonds.length() + _maturedBonds.length();
     }
 
     /// @inheritdoc IBondIssuer
+    /// @dev Internally, we first iterate through the active list and then move on to the mature list.
+    ///      [0, activeBonds.len) is NOT guaranteed to be ordered,
+    ///      [activeBonds.len, numBonds) will be ordered by increasing maturity date.
```

Comment:
not sure `_maturedBonds` is guaranteed to be ordered by increasing maturity date, even if it likely will in practice. 

example: `_activeBonds`, is, by maturity, [D, D+1, D+2], but after a `matureAll() after D, but before D+1, becomes [D+2, D+1]. Then a new bond gets added so its, [D+2, D+1, D+3]. If `matureAll()` does not get called again until D+3, the tail of `_matureBonds` would be [D+3, D+1, D+2]. 

unless we add `matureAll()` calls in, e.g. `issue()`, the only guarantee is that the function is poked externally and regularly

Reply:
yea we should assume that its not ordered